cwms_radar_api/src/main/java/cwms/radar/data/dao/TimeSeriesDaoImpl.java

```diff
@@ -0,0 +1,649 @@
+					.map(DSL::field).collect(
+					Collectors.toList());
+
+
+			// I want to select tsvView.asterisk but we are selecting from an inner select and
+			// even though the inner select selects tsvView.asterisk it isn't the same.
+			// So we will just select the fields we want.
+			// Unfortunately that means our results won't map into AV_TSV.AV_TSV
+			dsl.select(fields)
+					.from(innerSelect)
+					.where(field(tsvView.DATE_TIME.getName()).eq(innerSelect.field("max_date_time")))
+					.orderBy(field(tsView.ATTRIBUTE.getName()))
+					.forEach( jrecord -> {
+						RecentValue recentValue = buildRecentValue(tsvView, tsView, jrecord);
+						retval.add(recentValue);
+					});
+		}
+		return retval;
+	}
+
+	@NotNull
+	private TsvDquId buildDquId(AV_TSV_DQU tsvView, Record jrecord)
+	{
+		return new TsvDquId(jrecord.getValue(tsvView.OFFICE_ID.getName(), String.class),
+				jrecord.getValue(tsvView.TS_CODE.getName(), Long.class),
+				jrecord.getValue(tsvView.UNIT_ID.getName(), String.class),
+				jrecord.getValue(tsvView.DATE_TIME.getName(), Timestamp.class));
+	}
+
+	public void create(TimeSeries input)
+	{
+		dsl.connection(connection -> {
+			CwmsDbTs tsDao = CwmsDbServiceLookup.buildCwmsDb(CwmsDbTs.class, connection);
+
+			int utcOffsetMinutes = 0;
+			int intervalForward = 0;
+			int intervalBackward = 0;
+			boolean versionedFlag = false;
+			boolean activeFlag = true;
+			tsDao.createTsCodeBigInteger(connection, input.getOfficeId(), input.getName(), utcOffsetMinutes,
+					intervalForward, intervalBackward, versionedFlag, activeFlag);
+		});
+	}
+
+	public void store(TimeSeries input)
+	{
+		dsl.connection(connection -> {
+			store(connection, input.getOfficeId(), input.getName(), input.getUnits(), input.getValues());
+		});
+	}
+
+	public void update(TimeSeries input) throws SQLException
+	{
+		String name = input.getName();
+		if(!timeseriesExists(name)){
+			throw new SQLException("Cannot update a non-existant Timeseries. Create " + name + " first.");
+		}
+		dsl.connection(connection -> {
+			store(connection, input.getOfficeId(), name, input.getUnits(), input.getValues());
+		});
+	}
+
+	public void store(Connection connection, String officeId, String tsId, String units,
+						   List<TimeSeries.Record> values) throws SQLException
+	{
+		CwmsDbTs tsDao =  CwmsDbServiceLookup.buildCwmsDb(CwmsDbTs.class, connection);
+
+		final int count = values == null ? 0 : values.size();
+
+		final long[] timeArray = new long[count];
+		final double[] valueArray = new double[count];
+		final int[] qualityArray = new int[count];
+
+		if(values != null && !values.isEmpty())
+		{
+			Iterator<TimeSeries.Record> iter = values.iterator();
+			for(int i = 0; iter.hasNext(); i++)
+			{
+				TimeSeries.Record value = iter.next();
+				timeArray[i] = value.getDateTime().getTime();
+				valueArray[i] = value.getValue();
+				qualityArray[i] = value.getQualityCode();
+			}
+		}
+
+		final Timestamp versionDate = null;  // Where does this come from?
```

Comment:
This would be passed in by the user. Most commonly used for forecast data so the original isn't destroyed. this is a good place holder but I wonder if this logic should just be IN the first call to store that takes the TimeSeries object, as the versionDate will eventually be a non-required parameter of both that, and depending on how we choose to do things, the timeseries value elements. But I also haven't read further than this yet to know if this call is used else where.

Reply:
Instead of NULL we should probably have a constant, which would still be null, called NON_VERSIONED. It would make the section of code a little more clear.

Reply:
Sure. that sounds like a good idea.